x-pack/test_serverless/api_integration/test_suites/common/platform_security/authorization.ts

```diff
@@ -25,112 +25,93 @@ function collectSubFeaturesPrivileges(feature: KibanaFeatureConfig) {
 export default function ({ getService }: FtrProviderContext) {
   const log = getService('log');
   const svlCommonApi = getService('svlCommonApi');
-  const svlUserManager = getService('svlUserManager');
-  const supertestWithoutAuth = getService('supertestWithoutAuth');
-  const supertest = getService('supertest');
-  let roleAuthc: RoleCredentials;
-  let internalReqHeader: InternalRequestHeader;
+  const roleScopedSupertest = getService('roleScopedSupertest');
+  let supertestAdminWithCookieCredentials: SupertestWithRoleScopeType;
+  let supertestAdminWithApiKey: SupertestWithRoleScopeType;
 
   describe('security/authorization', function () {
     // see details: https://github.com/elastic/kibana/issues/192282
     this.tags(['failsOnMKI']);
     before(async () => {
-      roleAuthc = await svlUserManager.createM2mApiKeyWithRoleScope('admin');
-      internalReqHeader = svlCommonApi.getInternalRequestHeader();
+      supertestAdminWithCookieCredentials = await roleScopedSupertest.getSupertestWithRoleScope(
+        'admin',
+        {
+          useCookieHeader: true,
+          withInternalHeaders: true,
+        }
+      );
+      supertestAdminWithApiKey = await roleScopedSupertest.getSupertestWithRoleScope('admin', {
+        withInternalHeaders: true,
+      });
     });
     after(async () => {
-      await svlUserManager.invalidateM2mApiKeyWithRoleScope(roleAuthc);
+      await supertestAdminWithApiKey.destroy();
     });
     describe('route access', () => {
       describe('internal', () => {
         describe('disabled', () => {
           // Skipped due to change in QA environment for role management and spaces
           // TODO: revisit once the change is rolled out to all environments
           it.skip('get all privileges', async () => {
-            const { body, status } = await supertestWithoutAuth
-              .get('/api/security/privileges')
-              .set(svlCommonApi.getInternalRequestHeader())
-              .set(roleAuthc.apiKeyHeader);
+            const { body, status } = await supertestAdminWithApiKey.get('/api/security/privileges');
```

Comment:
Note: On serverless, these APIs are currently internal. I think it's fine to call them with an API key here,